### PR TITLE
[Rooms] Migrate test recordings to assets repo

### PR DESCRIPTION
**Pull request by Mrayyan**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
 import com.azure.core.http.rest.Response;
 import static org.junit.jupiter.api.Assertions.*;
 
-public class RoomsTestBase extends TestBase {
```
Comment (samvaity) [inline sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java:34]: I do see some redactions that need to be added. 
Currently, adding redactors directly to the record policy is not supported.
To add sanitizers follow - https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/core/azure-core-test#sanitize-secrets
needed below code changes
```
1) .httpClient(interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient)

2) builder.addPolicy(interceptorManager.getRecordPolicy());

3) FakeCredentials -> MockTokenCredential
````

File: sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java [hunk 2]
```diff
@@ -55,27 +52,33 @@ protected RoomsClientBuilder getRoomsClient(HttpClient httpClient) {
...
+            builder.addPolicy(interceptorManager.getRecordPolicy());
+        }
+
+        if (!interceptorManager.isLiveMode()) {
+            interceptorManager.addSanitizers(
+                    Arrays.asList(new TestProxySanitizer("$..id", null, "REDACTED", TestProxySanitizerType.BODY_KEY)));
         }
+
         return builder;
     }
```
Comment (samvaity) [inline sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java:66]: Need to add matchers
```
if (interceptorManager.isPLaybackMode()) {
interceptorManager.addMatchers(Arrays.asList(new BodilessMatcher(), new CustomMatcher().setHeadersKeyOnlyMatch("repeatability-first-sent", "repeatability-request-id", "x-ms-content-sha256", "x-ms-hmac-string-to-sign-base64")))
}
```

File: sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java [hunk 3]
```diff
@@ -160,7 +167,7 @@ private Mono<HttpResponse> logHeaders(String testName, HttpPipelineNextPolicy ne
         });
     }
 
-    static class FakeCredentials implements TokenCredential {
```
Comment (samvaity) [inline sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java:163]: revert and remove FakeCredentials,

File: sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java [hunk 4]
```diff
@@ -85,13 +91,20 @@ protected RoomsClientBuilder getRoomsClientWithConnectionString(HttpClient httpC
...
+            builder.addPolicy(interceptorManager.getRecordPolicy());
+        }
+        if (!interceptorManager.isLiveMode()) {
+            interceptorManager.addSanitizers(
+                    Arrays.asList(new TestProxySanitizer("$..id", null, "REDACTED", TestProxySanitizerType.BODY_KEY)));
         }
+
+        if (interceptorManager.isPlaybackMode()) {
+            interceptorManager.addMatchers(Arrays.asList(new BodilessMatcher(), new CustomMatcher().setHeadersKeyOnlyMatch(Arrays.asList("repeatability-first-sent", "repeatability-request-id", "x-ms-content-sha256", "x-ms-hmac-string-to-sign-base64"))));
+        }
```
Comment (samvaity) [inline sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java:106]: nit; Can add these to reusable functions.

File: sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java [hunk 5]
```diff
@@ -85,13 +91,20 @@ protected RoomsClientBuilder getRoomsClientWithConnectionString(HttpClient httpC
...
+                .httpClient(interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient);
 
         if (getTestMode() == TestMode.RECORD) {
-            List<Function<String, String>> redactors = new ArrayList<>();
-            redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
-            builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
+            builder.addPolicy(interceptorManager.getRecordPolicy());
+        }
+        if (!interceptorManager.isLiveMode()) {
+            interceptorManager.addSanitizers(
```
Comment (samvaity) [inline sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java:100]: nit; Can add these to reusable functions.

**Review by samvaity — APPROVED** [review]
🎉

**Review by joshfree — APPROVED** [review]

**Pull request merged**